packages/webhid/src/methods.ts

```diff
@@ -69,13 +69,28 @@ export async function setupXkeysPanel(browserDevice: HIDDevice): Promise<XKeys>
 		})
 	})
 
-	// Wait for the device to initialize:
-	try {
-		await xkeys.init()
-
-		return xkeys
-	} catch (e) {
-		await deviceWrap.close()
-		throw e
-	}
+	let alreadyRejected = false
+	return new Promise((resolve, reject) => {
+		const xkeysStopgapErrorHandler = (e: unknown) => {
+			if (alreadyRejected) {
+				console.error(`Xkeys: Error emitted after setup already rejected:`, e)
+				return
+			}
+			deviceWrap.close()
+				.then(() => reject())
+				.catch(reject)
+				.finally(() => (alreadyRejected = true))
+		}
+	    // Handle all error events until the instance is returned
+		xkeys.on('error', xkeysStopgapErrorHandler)
```

Comment:
we don't use `once` intentionally. Until we have resolved back the instance to the caller, we are responsible to keep catching errors that may be emitted.